MC test batch - yet another pack refreshed (applications-protocol-http) (#2640)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/applications-protocol-http.md

| WARNINGNUMERIC  | Warning threshold (Default: on total matching elements)                                                                                                                                                                                                                                                                                    |                                      |             |
| CRITICALNUMERIC | Critical threshold (Default: on total matching elements)                                                                                                                                                                                                                                                                                   |                                      |             |

> | WARNINGSTRING   | Warning threshold if the string match                                                                                                                                                                                                                                                                                                      |                                      |             |

Reviewer: matches what?

| EXTRAOPTIONS | Any extra option you may want to add to the command (E.g. a --verbose flag). All options are listed [here](#available-options) |                   |             |

</TabItem>
<TabItem value="HTTP-Soap-Content" label="HTTP-Soap-Content">

| Macro           | Description                                                                                                                                                                                                                                                                                                                                | Default value                        | Mandatory   |
|:----------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------|:-----------:|

> | URLPATH         | Set path to get Webpage (Default: '/')                                                                                                                                                                                                                                                                                                     | /                                    |             |

Reviewer: ```suggestion
| URLPATH         | Set path to get web page (Default: '/')                                                                                                                                                                                                                                                                                                     | /                                    |             |
```

| WARNINGNUMERIC  | Warning threshold (Default: on total matching elements)                                                                                                                                                                                                                                                                                    |                                      |             |
| CRITICALNUMERIC | Critical threshold (Default: on total matching elements)                                                                                                                                                                                                                                                                                   |                                      |             |

> | WARNINGSTRING   | Warning threshold if the string match                                                                                                                                                                                                                                                                                                      |                                      |             |

Reviewer: ```suggestion
| WARNINGSTRING   | Warning threshold if the string matches                                                                                                                                                                                                                                                                                                      |                                      |             |
```

Reviewer: matches what?

> | CRITICALSTRING  | Critical threshold if the string match                                                                                                                                                                                                                                                                                                     |                                      |             |

Reviewer: ```suggestion
| CRITICALSTRING  | Critical threshold if the string matches                                                                                                                                                                                                                                                                                                     |                                      |             |
```

Reviewer: matches what?

| EXTRAOPTIONS    | Any extra option you may want to add to the command (E.g. a --verbose flag). All options are listed [here](#available-options)                                                                                                                                                                                                                                        |                                      |             |

</TabItem>
</Tabs>

> 3. [Deploy the configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). The service appears in the list of services, and on page **Resources Status**. The command that is sent by the connector is displayed in the details panel of the service: it shows the values of the macros.

Reviewer: ```suggestion
3. [Deploy the configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). The service appears in the list of services, and on the **Resources Status** page. The command that is sent by the connector is displayed in the details panel of the service: it shows the values of the macros.
```

## How to check in the CLI that the configuration is OK and what are the main options for?

Once the plugin is installed, log into your Centreon poller's CLI using the
**centreon-engine** user account (`su - centreon-engine`). Test that the connector 
is able to monitor a resource using a command like this one (replace the sample values by yours):

```bash
/usr/lib/centreon/plugins/centreon_protocol_http.pl \
	--plugin=apps::protocols::http::plugin \
	--mode=response \

> 	--hostname=10.0.0.1 \

Reviewer: why 10.0.0.1?

#### Modes options

All available options for each service template are listed below:

<Tabs groupId="sync">
<TabItem value="HTTP-Expected-Content" label="HTTP-Expected-Content">

| Option               | Description                                                                                                                                                                                                                                                                                 |
|:---------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| --hostname           | IP Addr/FQDN of the Webserver host                                                                                                                                                                                                                                                          |

> | --port               | Port used by Webserver                                                                                                                                                                                                                                                                      |

Reviewer: ```suggestion
| --port               | Port used by the web server                                                                                                                                                                                                                                                                      |
```

| --method             | Specify http method used (Default: 'GET')                                                                                                                                                                                                                                                   |
| --proto              | Specify https if needed (Default: 'http')                                                                                                                                                                                                                                                   |
| --urlpath            | Set path to get Webpage (Default: '/')                                                                                                                                                                                                                                                      |

> | --credentials        | Specify this option if you access webpage with authentication                                                                                                                                                                                                                               |

Reviewer: ```suggestion
| --credentials        | Specify this option if you are accessing  a web page with authentication                                                                                                                                                                                                                               |
```

| --username           | Specify username for authentication (Mandatory if --credentials is specified)                                                                                                                                                                                                               |
| --password           | Specify password for authentication (Mandatory if --credentials is specified)                                                                                                                                                                                                               |
| --basic              | Specify this option if you access webpage over basicauthentication and don't want a '401 UNAUTHORIZED' error to be logged on your webserver.  Specify this option if you access webpage over hidden basic authentication or you'll get a '404 NOT FOUND' error.  (Use with --credentials)   |

> | --ntlmv2             | Specify this option if you access webpage over ntlmv2 authentication (Use with --credentials and --port options)                                                                                                                                                                            |

Reviewer: ```suggestion
| --ntlmv2             | Specify this option if you are accessing a web page using ntlmv2 authentication (Use with --credentials and --port options)                                                                                                                                                                            |
```

| --timeout            | Threshold for HTTP timeout (Default: 5)                                                                                                                                                                                                                                                     |
| --no-follow          | Do not follow http redirect                                                                                                                                                                                                                                                                 |

> | --cert-file          | Specify certificate to send to the webserver                                                                                                                                                                                                                                                |

Reviewer: ```suggestion
| --cert-file          | Specify certificate to send to the web server                                                                                                                                                                                                                                                |
```

> | --key-file           | Specify key to send to the webserver                                                                                                                                                                                                                                                        |

Reviewer: ```suggestion
| --key-file           | Specify key to send to the web server                                                                                                                                                                                                                                                        |
```

> | --cacert-file        | Specify root certificate to send to the webserver                                                                                                                                                                                                                                           |

Reviewer: ```suggestion
| --cacert-file        | Specify root certificate to send to the web server                                                                                                                                                                                                                                           |
```

| --get-param          | Set GET params (Multiple option. Example: --get-param='key=value')                                                                                                                                                                                                                          |
| --post-param         | Set POST params (Multiple option. Example: --post-param='key=value')                                                                                                                                                                                                                        |
| --cookies-file       | Save cookies in a file (Example: '/tmp/lwp\_cookies.dat')                                                                                                                                                                                                                                   |

> | --extracted-pattern  | Set pattern to extracted a number (used --warning-extracted and --critical-extracted option).                                                                                                                                                                                               |

Reviewer: ```suggestion
| --extracted-pattern  | Set pattern to extract a number (use --warning-extracted and --critical-extracted options).                                                                                                                                                                                               |
```

| --unknown-status     | Warning threshold for http response code (Default: '%{http\_code} \< 200 or %{http\_code} \>= 300')                                                                                                                                                                                         |
| --warning-status     | Warning threshold for http response code                                                                                                                                                                                                                                                    |
| --critical-status    | Critical threshold for http response code                                                                                                                                                                                                                                                   |

> | --warning-time       | Warning threshold in seconds (Webpage response time)                                                                                                                                                                                                                                        |

Reviewer: ```suggestion
| --warning-time       | Warning threshold in seconds (web page response time)                                                                                                                                                                                                                                        |
```

</TabItem>
<TabItem value="HTTP-Json-Content" label="HTTP-Json-Content">

| Option                    | Description                                                                                                                                                                                                                                                                                                                                  |
|:--------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| --data                    | Set file with JSON request                                                                                                                                                                                                                                                                                                                   |

> | --lookup                  | What to lookup in JSON response (JSON XPath string) (can be multiple) See: http://goessner.net/articles/JsonPath/                                                                                                                                                                                                                            |

Reviewer: ```suggestion
| --lookup                  | What to look up in JSON response (JSON XPath string) (can be multiple) See: http://goessner.net/articles/JsonPath/                                                                                                                                                                                                                            |
```

| --lookup-perfdatas-nagios | Take perfdatas from the JSON response (JSON XPath string) Chain must be formated in Nagios format. Ex : "rta=10.752ms;50.000;100.000;0; pl=0%;20;40;; rtmax=10.802ms;;;;"      FORMAT OPTIONS:                                                                                                                                               |

> | --format-lookup           | Take the output message from the JSON response (JSON XPath string) Override all the format options but substitute are still applied.                                                                                                                                                                                                         |

Reviewer: substitute?

> | --format-lookup           | Take the output message from the JSON response (JSON XPath string) Override all the format options but substitute are still applied.                                                                                                                                                                                                         |

Author: ```suggestion
| --format-lookup           | Take the output message from the JSON response (JSON XPath string) Override all the format options but substitute are still applied.                                                                                                                                                                                                         |
```
```suggestion
| --format-lookup           | Take the output message from the JSON response (JSON XPath string). Overrides all the format options but substitutions are still applied.                                                                                                                                                                                                         |
```

| --warning-numeric         | Warning threshold (Default: on total matching elements)                                                                                                                                                                                                                                                                                      |
| --critical-numeric        | Critical threshold (Default: on total matching elements)                                                                                                                                                                                                                                                                                     |
| --threshold-value         | Which value to use (Default: 'count') Can be: 'values' (only check numeric values)                                                                                                                                                                                                                                                           |

> | --warning-string          | Warning threshold if the string match                                                                                                                                                                                                                                                                                                        |

Reviewer: ```suggestion
| --warning-string          | Warning threshold if the string matches                                                                                                                                                                                                                                                                                                        |
```

Reviewer: matches what?

> | --critical-string         | Critical threshold if the string match                                                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| --critical-string         | Critical threshold if the string matches                                                                                                                                                                                                                                                                                                       |
```

Reviewer: matches what?

> | --unknown-string          | Threshold unknown if the string match                                                                                                                                                                                                                                                                                                        |

Reviewer: ```suggestion
| --unknown-string          | Unknown threshold if the string matches                                                                                                                                                                                                                                                                                                       |
```

Reviewer: matches what?

| --warning-time            | Warning threshold in seconds of webservice response time                                                                                                                                                                                                                                                                                     |
| --critical-time           | Critical threshold in seconds of webservice response time      HTTP OPTIONS:                                                                                                                                                                                                                                                                 |
| --hostname                | IP Addr/FQDN of the Webserver host                                                                                                                                                                                                                                                                                                           |

> | --port                    | Port used by Webserver                                                                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| --port                    | Port used by web server                                                                                                                                                                                                                                                                                                                       |
```

| --credentials             | Specify this option if you access webpage with authentication                                                                                                                                                                                                                                                                                |
| --username                | Specify username for authentication (Mandatory if --credentials is specified)                                                                                                                                                                                                                                                                |
| --password                | Specify password for authentication (Mandatory if --credentials is specified)                                                                                                                                                                                                                                                                |

> | --basic                   | Specify this option if you access webpage over basicauthentication and don't want a '401 UNAUTHORIZED' error to be logged on your webserver.  Specify this option if you access webpage over hidden basic authentication or you'll get a '404 NOT FOUND' error.  (Use with --credentials)                                                    |

Reviewer: ```suggestion
| --basic                   | Specify this option if you are accessing a web page using basic authentication and don't want a '401 UNAUTHORIZED' error to be logged on your web server.  Specify this option if you are accessing a web page using hidden basic authentication or you'll get a '404 NOT FOUND' error.  (Use with --credentials)                                                    |
```

> | --ntlmv2                  | Specify this option if you access webpage over ntlmv2 authentication (Use with --credentials and --port options)                                                                                                                                                                                                                             |

Reviewer: ```suggestion
| --ntlmv2                  | Specify this option if you are accessing a web page using ntlmv2 authentication (Use with --credentials and --port options)                                                                                                                                                                                                                             |
```

| --timeout                 | Threshold for HTTP timeout (Default: 10)                                                                                                                                                                                                                                                                                                     |

> | --cert-file               | Specify certificate to send to the webserver                                                                                                                                                                                                                                                                                                 |

Reviewer: ```suggestion
| --cert-file               | Specify certificate to send to the web server                                                                                                                                                                                                                                                                                                 |
```

| --port                   | Port used by Webserver                                                                                                                                                                                                                                                                      |
| --method                 | Specify http method used (Default: 'GET')                                                                                                                                                                                                                                                   |
| --proto                  | Specify https if needed (Default: 'http')                                                                                                                                                                                                                                                   |

> | --urlpath                | Set path to get webpage (Default: '/')                                                                                                                                                                                                                                                      |

Reviewer: ```suggestion
| --urlpath                | Set path to get web page (Default: '/')                                                                                                                                                                                                                                                      |
```

> | --credentials            | Specify this option if you access webpage with authentication                                                                                                                                                                                                                               |

Reviewer: ```suggestion
| --credentials            | Specify this option if you are accessing a web page with authentication                                                                                                                                                                                                                               |
```

| --username               | Specify username for authentication (Mandatory if --credentials is specified)                                                                                                                                                                                                               |
| --password               | Specify password for authentication (Mandatory if --credentials is specified)                                                                                                                                                                                                               |

> | --basic                  | Specify this option if you access webpage over basicauthentication and don't want a '401 UNAUTHORIZED' error to be logged on your webserver.  Specify this option if you access webpage over hidden basic authentication or you'll get a '404 NOT FOUND' error.  (Use with --credentials)   |

Reviewer: ```suggestion
| --basic                  | Specify this option if you are accessing a web page using basic authentication and don't want a '401 UNAUTHORIZED' error to be logged on your web server.  Specify this option if you are accessing a web page using hidden basic authentication or you'll get a '404 NOT FOUND' error.  (Use with --credentials)   |
```

| --ntlmv2                 | Specify this option if you access webpage over ntlmv2 authentication (Use with --credentials and --port options)                                                                                                                                                                            |
| --timeout                | Threshold for HTTP timeout (Default: 5)                                                                                                                                                                                                                                                     |
| --no-follow              | Do not follow http redirect                                                                                                                                                                                                                                                                 |

> | --cert-file              | Specify certificate to send to the webserver                                                                                                                                                                                                                                                |

Reviewer: ```suggestion
| --cert-file              | Specify certificate to send to the web server                                                                                                                                                                                                                                                |
```

> | --key-file               | Specify key to send to the webserver                                                                                                                                                                                                                                                        |

Reviewer: ```suggestion
| --key-file               | Specify key to send to the web server                                                                                                                                                                                                                                                        |
```

> | --cacert-file            | Specify root certificate to send to the webserver                                                                                                                                                                                                                                           |

Reviewer: ```suggestion
| --cacert-file            | Specify root certificate to send to the web server                                                                                                                                                                                                                                           |
```

|:-------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| --service-soap     | Service Soap Action (Required)                                                                                                                                                                                                                                                                                                               |
| --data             | Set file with SOAP request (Required)                                                                                                                                                                                                                                                                                                        |

> | --lookup           | What to lookup in XML response (XPath string) (can be multiple)      FORMAT OPTIONS:                                                                                                                                                                                                                                                         |

Reviewer: ```suggestion
| --lookup           | What to look up in XML response (XPath string) (can be multiple)      FORMAT OPTIONS:                                                                                                                                                                                                                                                         |
```

| --warning-numeric  | Warning threshold (Default: on total matching elements)                                                                                                                                                                                                                                                                                      |
| --critical-numeric | Critical threshold (Default: on total matching elements)                                                                                                                                                                                                                                                                                     |
| --threshold-value  | Which value to use (Default: 'count') Can be: 'values' (only check numeric values)                                                                                                                                                                                                                                                           |

> | --warning-string   | Warning threshold if the string match                                                                                                                                                                                                                                                                                                        |

Reviewer: ```suggestion
| --warning-string   | Warning threshold if the string matches                                                                                                                                                                                                                                                                                                        |
```

Reviewer: matches what?

> | --critical-string  | Critical threshold if the string match                                                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| --critical-string  | Critical threshold if the string matches                                                                                                                                                                                                                                                                                                       |
```

Reviewer: matches what?

| --warning-time     | Warning threshold in seconds of webservice response time                                                                                                                                                                                                                                                                                     |
| --critical-time    | Critical threshold in seconds of webservice response time      HTTP OPTIONS:                                                                                                                                                                                                                                                                 |

> | --hostname         | IP Addr/FQDN of the Webserver host                                                                                                                                                                                                                                                                                                           |

Reviewer: ```suggestion
| --hostname         | IP address/FQDN of the web server's host                                                                                                                                                                                                                                                                                                           |
```

> | --port             | Port used by Webserver                                                                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| --port             | Port used by web server                                                                                                                                                                                                                                                                                                                       |
```

| --basic            | Specify this option if you access webpage over basicauthentication and don't want a '401 UNAUTHORIZED' error to be logged on your webserver.  Specify this option if you access webpage over hidden basic authentication or you'll get a '404 NOT FOUND' error.  (Use with --credentials)                                                    |
| --ntlmv2           | Specify this option if you access webpage over ntlmv2 authentication (Use with --credentials and --port options)                                                                                                                                                                                                                             |
| --timeout          | Threshold for HTTP timeout (Default: 10)                                                                                                                                                                                                                                                                                                     |

> | --cert-file        | Specify certificate to send to the webserver                                                                                                                                                                                                                                                                                                 |

Reviewer: ```suggestion
| --cert-file        | Specify certificate to send to the web server                                                                                                                                                                                                                                                                                                 |
```

> | --key-file         | Specify key to send to the webserver                                                                                                                                                                                                                                                                                                         |

Reviewer: ```suggestion
| --key-file         | Specify key to send to the web server                                                                                                                                                                                                                                                                                                         |
```

> | --cacert-file      | Specify root certificate to send to the webserver                                                                                                                                                                                                                                                                                            |

Reviewer: ```suggestion
| --cacert-file      | Specify root certificate to send to the web server                                                                                                                                                                                                                                                                                            |
```
